notebooks/1000_run-workflow-ar6-style.py

```diff
@@ -0,0 +1,273 @@
+# ### Down-select scenarios
+
+# %%
+# Randomly select some scenarios
+# (this is how I generated the hard-coded values in the next cell).
+base = pre_pre_processed.pix.unique(["model", "scenario"]).to_frame(index=False)
+base["scenario_group"] = base["scenario"].apply(lambda x: x.split("-")[-1].split("_")[0].strip())
+
+selected_scenarios_l = []
+selected_models = []
+for scenario_group, sdf in base.groupby("scenario_group"):
+    options = sdf.index.values.tolist()
+    random.shuffle(options)
+
+    n_selected = 0
+    for option_loc in options:
+        selected_model = sdf.loc[option_loc, :].model
+        if selected_model not in selected_models:
+            selected_scenarios_l.append(sdf.loc[option_loc, :])
+            selected_models.append(selected_model)
+            n_selected += 1
+            if n_selected >= 2:  # noqa: PLR2004
+                break
+
+    else:
+        if n_selected >= 1:
+            selected_scenarios_l.append(sdf.loc[option_loc, :])
+            selected_models.append(selected_model)
+        else:
+            selected_scenarios_l.append(sdf.loc[option_loc, :])
+            selected_models.append(selected_model)
+
+            option_loc = options[-2]
+            selected_model = sdf.loc[option_loc, :].model
+            selected_scenarios_l.append(sdf.loc[option_loc, :])
+            selected_models.append(selected_model)
+
+selected_scenarios = pd.concat(selected_scenarios_l, axis="columns").T
+selected_scenarios_idx = selected_scenarios.set_index(["model", "scenario"]).index
+selected_scenarios
+
+# %%
+selected_scenarios_idx = pd.MultiIndex.from_tuples(
+    (
+        ("MESSAGEix-GLOBIOM 2.1-M-R12", "SSP5 - High Emissions"),
+        ("IMAGE 3.4", "SSP5 - High Emissions"),
+        ("AIM 3.0", "SSP2 - Medium-Low Emissions"),
+        ("WITCH 6.0", "SSP2 - Low Emissions"),
+        ("REMIND-MAgPIE 3.4-4.8", "SSP2 - Low Overshoot_b"),
+        ("MESSAGEix-GLOBIOM-GAINS 2.1-M-R12", "SSP5 - Low Overshoot"),
+        ("COFFEE 1.5", "SSP2 - Medium Emissions"),
+        ("GCAM 7.1 scenarioMIP", "SSP2 - Medium Emissions"),
+        ("IMAGE 3.4", "SSP2 - Very Low Emissions"),
+        ("MESSAGEix-GLOBIOM-GAINS 2.1-M-R12", "SSP1 - Very Low Emissions"),
+    ),
+    name=["model", "scenario"],
+)
+scenarios_run = pre_pre_processed[pre_pre_processed.index.isin(selected_scenarios_idx)]
+
+# scenarios_run = pre_pre_processed.loc[pix.ismatch(scenario="*Very Low*")]
+
+# %%
+# To run all, just uncomment the below
+scenarios_run = pre_pre_processed
+
+# %%
+scenarios_run.pix.unique(["model", "scenario"]).to_frame(index=False)
+
+# %%
+res = run_ar6_workflow(
+    input_emissions=scenarios_run,
+    magicc_exe_path=magicc_exe_path,
+    magicc_prob_distribution_path=magicc_prob_distribution_path,
+    batch_size_scenarios=batch_size_scenarios,
+    scm_output_variables=scm_output_variables,
+    scm_results_db=scm_results_db,
+    n_processes=n_processes,
+    run_checks=False,  # TODO: turn this back on
```

Comment:
Same question as https://github.com/iiasa/emissions_harmonization_historical/pull/58/files#r1937460450

Reply:
No it's because there are no checks implemented 🙃